SERVER/CONTROLLERS/userAcc_Controler.js

```diff
@@ -1,71 +1,59 @@
         if(create_acc_validation.error){
             const signup_errors=[];
-            for(let index=0; index<create_acc_validation.error.details.length; index++){
-                signup_errors.push(create_acc_validation.error.details[index].message.split('"').join(" "));
+            for(let i=0; i<create_acc_validation.error.details.length; i++){
+                signup_errors.push(create_acc_validation.error.details[i].message.split('"').join(" "));
             }
             return res.status(400).send({
                 status: 400,
                 error: signup_errors[0]
             });
         }else{
-            const all_user_accounts= accounts.AllAccounts;
-            const acc_exists= all_user_accounts.find(acc=>acc.email===email);
-            if(acc_exists){
-                return res.status(409).json({
-                    status:409,
-                    message: `This account already exists.Try another one.`
-                });
-            }else{
-                const createdAccount= accounts.createAccount(req.body);
-                const user_token= {
-                    id: createdAccount.id,
-                    firstName: createdAccount.firstName,
-                    lastName: createdAccount.lastName,
-                    email: createdAccount.email,
-                    address: createdAccount.address,
-                    bio: createdAccount.bio,
-                    occupation: createdAccount.occupation,
-                    expertise: createdAccount.expertise,
-                    isAdmin: createdAccount.isAdmin,
-                    isAmentor: createdAccount.isAmentor
-                };
-
-                const token= jwt.sign(user_token, user_secret, { expiresIn: "1d" });
-                res.header('x-auth-token', token);
-
-                return res.status(201).json({
-                    status: 201,
-                    message : 'User created successfully',
-                    data: {
-                        token: token,
-                        id: createdAccount.id,
-                        firstName: firstName,
-                        lastName: lastName,
-                        email: email,
-                        address: address,
-                        bio: bio,
-                        occupation: occupation,
-                        expertise: expertise,
-                        isAdmin: createdAccount.isAdmin,
-                        isAmentor: createdAccount.isAmentor
+            pool.connect((err, client, done) => {
+                const values = [email.trim()];
+                client.query(selectQuery, values, (error, result) => {
+                    if (result.rows[0]) {
+                        return res.status(409).send({
+                            status: 409,
+                            error: `This account already exists.Try another one.`
+                        });
+                    }else {
+                        pool.connect((err, client, done) => {
```

Comment:
I don't understand why you are opening the connection twice. Why not use the same client to check if the user exists & then add the user?

Reply:
God morning Shalon, I am going to solve it.